build/tools/go.mod

```diff
@@ -9,9 +9,9 @@ require (
 	github.com/golangci/golangci-lint v1.52.2
 	github.com/jstemmer/go-junit-report v1.0.0
 	google.golang.org/grpc/cmd/protoc-gen-go-grpc v1.3.0
-	google.golang.org/protobuf v1.30.0
+	google.golang.org/protobuf v1.34.2
 	mvdan.cc/gofumpt v0.5.0
-	sigs.k8s.io/controller-tools v0.12.0
+	sigs.k8s.io/controller-tools v0.16.5
```

Comment:
I don't think we can do this, [controller-runtime versions target specific k8s versions](https://github.com/kubernetes-sigs/controller-tools?tab=readme-ov-file#compatibility) and we need to freeze for 1.27 compatibility

Reply:
The problem is that controller-tools at 12.0 is not usable

Reply:
it panics: https://github.com/Azure/azure-container-networking/actions/runs/11486652516/job/32250025824

Reply:
Per `controller-tools` 
> Every minor version of controller-tools has been tested with a specific minor version of client-go. A controller-tools minor version may be compatible with other client-go minor versions, but this is by chance and neither supported nor tested. In general, we create one minor version of controller-tools for each minor version of client-go and other k8s.io/* dependencies.

This means that by chance it was working for 1.21 when they only meant for it to work for 1.20. We either rollback to 1.21 for our build tools, make a controller tools specific go.mod that will use go 1.20 or 1.21, or bump the controller-tools dep to v0.14.0+.

All have drawbacks...

Reply:
For more context, this only [broke for us once we updated](https://github.com/Azure/azure-container-networking/actions/runs/11153809821/job/31032529322?pr=3046) to 1.22.7 in #3046.